### ISSUE TITLE
Runnable issue of quantitive evaluation

### ISSUE THREAD

**Issue by aiwen7** [ChristopherLu/milliEgo#3]
Hello, 
Thank you for providing the code for the interesting project. 
I have an issue when running "os_evaluate_robot.py" that allows quantitive evaluation. It seems that some CSV files (e.g., true_delta_gmapping.csv) and some detailed configurations in "config.yaml" (e.g., cfg['dataset_creation']['dataroot']) are missing in the repository. Or have I missed some critical information in the readme file?
I would appreciate it very much if you could check them out and make "os_evaluate_robot.py" runnable.

**Comment by li-peize** [comment]
I see the problem. This repo is a slim down version of a large project. Do you still need our evaluation?

**Comment by li-peize** [comment]
The csv file is generated by os_evaluate_robot.py by calling composing_delta_pose.py. cfg['dataset_creation']['dataroot'] simply tells the scripts where to put them. I'll fix this problem soon. Sorry for the confusion.

**Comment by aiwen7** [comment]
Thanks for your response! Yes, we need the evaluation. Just as we expressed in previous emails, we are interested in reproducing the results reported in Table 1 of the paper. And we have three questions overall.

First, we trained the official code under default configuration with the pre-trained CNN model and from scratch (without pre-trained models), respectively. In our case, we took out nine training sequences as the validation set. We found that the model with pre-trained CNN has a better performance than the model from scratch in terms of visualization and quantification results. But both are generally worse than the provided model, i.e., '140'. Therefore, we want to know how the pre-trained models (i.e., 'cnn.h5' and '140') are trained and generated.

Second, we would like to know whether the results in Table 1 correspond to an evaluation of one exemplar testing trajectory (one specific testing sequence) or the average of all eight testing sequences given in the dataset. In our case, we directly evaluated the ATE-3D metrics with the provided model named '140'. We found the results neither on each testing sequence nor the average could precisely match the results in Table 1. 

Third, since 'os_evaluate_robot.py' is not runnable, we quantified the ATE-3D metrics by the following steps:
1) Run 'test_trajectory.py' with the trained model;
2) Run 'eval/convert_pose_to_quat.py' to generate the Ground Truth and the Prediction Path files;
3) Run 'eval/evaluate_std_ate.py' to generate the ATD-3D metrics;
We would like to know whether the above steps are correct for ATE-2D and ATE-3D metrics evaluation. And we noticed that you are fixing the problem with the evaluation. Thanks for helping!

**Comment by li-peize** [comment]
Hi, sorry for taking some time. Issue about os_evaluate_robot.py has been fixed. Thanks for bringing the issues to our attention.

Here is results from my running: [Google Drive File](https://drive.google.com/file/d/1Z4lPEpbtsw-A8WzyFwWoUwhW5uXHOOvV/view?usp=sharing)

I'll close this issue for now. If you find any further problems, you are welcome to open another issue.
